Source/Csla/Core/ContextDictionary.cs

```diff
@@ -76,25 +76,34 @@ void IMobileObject.SetChildren(SerializationInfo info, MobileFormatter formatter
     /// <inheritdoc cref="System.Collections.IDictionary.IsReadOnly"/>
     public bool IsReadOnly
     {
-      get => ((IDictionary)this).IsReadOnly;
+      get => false;
     }
 
     /// <inheritdoc cref="System.Collections.IDictionary.IsFixedSize"/>
     public bool IsFixedSize
     {
-      get => ((IDictionary)this).IsFixedSize;
+      get => false;
     }
 
     /// <inheritdoc cref="System.Collections.IDictionary.Add(object, object?)"/>
     public void Add(object key, object value)
     {
-      ((IDictionary)this).Add(key, value);
+      bool added = TryAdd(key, value);
+      if (!added)
+      {
+        throw new ArgumentException("An item with the same key has already been added.");
```

Comment:
1. Is this the right exception to throw? It seems maybe wrong?
2. English text can't be hard-coded. Either the text needs to come from the Resources file or avoid using text at all.

Reply:
That is the exception the `IDictionary` Add method raises for an already existing key if you peek at the documentation.

Reply:
OK, then it must be the right exception type 😄 

We still need the text to be in the resx file and come from the Resources class.

Reply:
Changed to use resource